### PR TITLE
Changes from Yankes review of RC1

### PR DESCRIPTION
**Pull request by memmaker**

### CODE REVIEW

File: src/Battlescape/SkillMenuState.cpp [hunk 1]
```diff
@@ -73,25 +67,37 @@ SkillMenuState::SkillMenuState(BattleAction *action, int x, int y) : ActionMenuS
 		Options::keyBattleActionItem2,
 		Options::keyBattleActionItem1
 	};
-	auto soldier = _action->actor->getGeoscapeSoldier();
+
 	for (auto skill : soldier->getRules()->getSkills())
 	{
-		if (!hotkeys.empty()
-			&& soldierHasAllRequiredBonusesForSkill(soldier, skill)
+		if (soldierHasAllRequiredBonusesForSkill(soldier, skill)
```
Comment (Yankes) [inline src/Battlescape/SkillMenuState.cpp]: we need break condition for `id == std::size(_actionMenu)`

File: src/Battlescape/SkillMenuState.cpp [hunk 2]
```diff
@@ -209,8 +215,7 @@ void SkillMenuState::btnActionMenuItemClick(Action *action)
 
 		if (!continueAction || _action->type == BA_NONE)
 		{
-			_action->type = BA_NONE;
-			_action->targeting = false;
+			_action->clear();
```
Comment (Yankes) [inline src/Battlescape/SkillMenuState.cpp]: Could you copy this change to other places too? If some code reset `type` and `targeting` it should be replaced by `clear()`

File: src/Battlescape/SkillMenuState.cpp (hunk 1 above)
Comment (memmaker) [reply src/Battlescape/SkillMenuState.cpp]: I don't get this, what do you mean here?

**Comment by MeridianOXC** [comment]
partially merged, and added afterLoad for skills... rest can be done in separate commits

**Pull request closed without merging**